File: Module 1/JS I - Fundamentals/homework_2_due_06.01.2017/problem_06/QuadraticEquation.js
// Problem 06: Quadratic Equation - BGCoder format
function solve(args) {
	var a = +args[0],
		b = +args[1],
		c = +args[2],
		determinant = +(b * b - 4 * a * c);		
		
	if (determinant < 0) {
		var result = ('no real roots');		
		return result;
	}
	
	if (determinant == 0) {
		var result = ('x1=x2=' + (-b / (2 * a)).toFixed(2));		
		return result;
	}
	
	if (determinant > 0) {
		var x1 = (((Math.sqrt(determinant)) - b) / (2 * a)).toFixed(2),
			x2 = (((-Math.sqrt(determinant)) - b) / (2 * a)).toFixed(2);
				
		if (x1 > x2) {
			if ((x1 > 0) || (x2 > 0)) {
				var temp = x1,
					x1 = x2,
					x2 = temp;
			}
		}
		
		if ((x1 < 0) && (x2 < 0)) {
			var temp1 = x1,
				x1 = x2,
				x2 = temp1;
		}
		
		var result = ('x1=' + x1 + '; x2=' + x2);		
		return result;
	}
}

File: Module 1/C# I - Fundamentals/homework_4_c_sharp_due_28.10.2016/06. Quadratic Equation/QuadraticEquation.cs
/**
Quadratic Equation
Description

Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).
Input

    On the first three lines, you will receive the coefficients a, b, and c, each on a separate line in the same order

Output

    If two different real roots exist, print them on two separate lines
        Print the smaller root on the first line
    If only one real root exists, print it on the only output line
    If no real root exists, print the string "no real roots"
    The roots, should they exist, must be printed with precision exactly two digits after the floating point

Constraints

    The input will always consist of valid real numbers in the range [-1000, 1000] and will follow the described format
    Time limit: 0.1s
    Memory limit: 16MB

Sample tests
Input 	Output
2
5
-3 	-3.00
0.50
-1
3
0 	0.00
3.00
-0.5
4
-8 	4.00
5
2
8 	no real roots
**/

using System;
using System.Globalization;
using System.Threading;

class QuadraticEquation
{
    static void Main()
    {
        Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;
        double a = double.Parse(Console.ReadLine());
        double b = double.Parse(Console.ReadLine());
        double c = double.Parse(Console.ReadLine());
        double x1 = (-b + Math.Sqrt(b * b - 4 * a * c)) / (2 * a);
        double x2 = (-b - Math.Sqrt(b * b - 4 * a * c)) / (2 * a);

        string x1Result = x1.ToString("0.00");
        string x2Result = x2.ToString("0.00");

        if (x1Result == "NaN" && x2Result == "NaN")
        {
            Console.WriteLine("no real roots");
        }
        else if (x1 == x2)
        {
            Console.WriteLine(x1Result);
        }
        else
        {
            if (x1 < x2)
            {
                Console.WriteLine(x1Result);
                Console.WriteLine(x2Result);
            }
            else
            {
                Console.WriteLine(x2Result);
                Console.WriteLine(x1Result);
            }
        }
    }
}